stl/inc/xutility

```diff
@@ -3918,76 +3918,91 @@ _NODISCARD constexpr auto _To_address(const _Iter& _Val) noexcept {
 }
 #endif // ^^^ !defined(__cpp_lib_concepts) ^^^
 
+template <class _Iter>
+_NODISCARD constexpr auto _To_address(const move_iterator<_Iter>& _Val) noexcept {
+    return _To_address(_Val.base());
+}
+
 // _Iterators_are_contiguous<_Iter1, _Iter2> reports whether both iterators are known to be contiguous.
 
 template <class _Iter1, class _Iter2>
 _INLINE_VAR constexpr bool _Iterators_are_contiguous =
     _Iterator_is_contiguous<_Iter1>&& _Iterator_is_contiguous<_Iter2>;
 
-template <class _Source, class _Dest>
-struct _Ptr_cat_helper {
-    using _USource                        = _Unwrap_enum_t<_Source>;
-    using _UDest                          = _Unwrap_enum_t<_Dest>;
-    static constexpr bool _Really_trivial = conjunction_v<
-        bool_constant<sizeof(_USource) == sizeof(_UDest) && is_same_v<bool, _USource> == is_same_v<bool, _UDest>>,
-        is_integral<_USource>, is_integral<_UDest>>;
-    static constexpr bool _Trivially_copyable = _Really_trivial;
-};
+template <class _Iter>
+_INLINE_VAR constexpr bool _Iterator_is_volatile = is_volatile_v<remove_reference_t<_Iter_ref_t<_Iter>>>;
 
-template <class _Elem>
-struct _Ptr_cat_helper<_Elem, _Elem> { // determines _Ptr_cat's result when the types are the same
-    static constexpr bool _Really_trivial     = is_trivial_v<_Elem>;
-    static constexpr bool _Trivially_copyable = is_trivially_copyable_v<_Elem>;
+template <class _Source, class _Dest>
+_INLINE_VAR constexpr bool _Is_pointer_address_convertible = is_void_v<_Source> || is_void_v<_Dest>
+            // NOTE: is_same_v is required for function pointers to work
+            || is_same_v<remove_cv_t<_Source>, remove_cv_t<_Dest>>
+#ifdef __cpp_lib_is_pointer_interconvertible
+            || is_pointer_interconvertible_base_of_v<_Dest, _Source>
+#endif // __cpp_lib_is_pointer_interconvertible
+        ;
+
+template <class _Source, class _Dest, class _SourceRef, class _DestRef>
+struct _Trivial_cat {

```

Comment:
I am not through it but could we move away from the structs and use simply variable templates?

Reply:
I think it would be less readable and less organized if we used variable templates. Grouping it in a struct seems cleaner.